fuzzamoto-ir/src/generators/bloom_filter.rs

```diff
@@ -0,0 +1,232 @@
+
+#[derive(Debug, Default)]
+pub struct BloomFilterLoadGenerator;
+
+fn init_filter<R: RngCore>(builder: &mut ProgramBuilder, rng: &mut R) -> IndexedVariable {
+    let ntweak = rng.gen_range(0..=u32::MAX);
+    let nflags = rng.gen_range(0..=2);
+
+    // let size = 0; triggers CVE-2013-5700
+    let size = rng.gen_range(0..MAX_BLOOM_FILTER_SIZE) as usize;
+    // Empty filter or random bytes
+    let mut bytes = vec![0; size];
+    if rng.gen_bool(0.5) {
+        rng.fill_bytes(bytes.as_mut_slice());
+    };
+    let nhash_funcs = rng.gen_range(1..MAX_HASH_FUNCS) as u32;
+
+    builder
+        .append(Instruction {
+            inputs: vec![],
+            operation: Operation::LoadFilterLoad {
+                filter: bytes,
+                hash_funcs: nhash_funcs,
+                tweak: ntweak,
+                flags: nflags,
+            },
+        })
+        .expect("Inserting LoadFilterLoad should always succeed")
+        .pop()
+        .expect("LoadFilterLoad should always produce a var")
+}
+
+fn populate_filter_from_txdata(
+    builder: &mut ProgramBuilder,
+    filter_var: IndexedVariable,
+    txs: &[IndexedVariable],
+    txos: &[IndexedVariable],
+) -> IndexedVariable {
+    let mut_filter_var = builder
+        .append(Instruction {
+            inputs: vec![filter_var.index],
+            operation: Operation::BeginBuildFilterLoad,
+        })
+        .expect("Inserting BeginBuildFilterLoad should always succeed")
+        .pop()
+        .expect("BeginBuildFilterLoad should always produce a var");
+
+    for tx in txs {
+        builder
+            .append(Instruction {
+                inputs: vec![mut_filter_var.index, tx.index],
+                operation: Operation::AddTxToFilter,
+            })
+            .expect("Inserting AddTxToFilter should always succeed");
+    }
+    for txo in txos {
+        builder
+            .append(Instruction {
+                inputs: vec![mut_filter_var.index, txo.index],
+                operation: Operation::AddTxoToFilter,
+            })
+            .expect("Inserting AddTxoToFilter should always succeed");
+    }
+
+    builder
+        .append(Instruction {
+            inputs: vec![mut_filter_var.index],
+            operation: Operation::EndBuildFilterLoad,
+        })
+        .expect("Inserting EndBuildFilterLoad should always succeed")
+        .pop()
+        .expect("EndBuildFilterLoad should always produce a var")
+}
+
+impl<R: RngCore> Generator<R> for BloomFilterLoadGenerator {
+    fn generate(&self, builder: &mut ProgramBuilder, rng: &mut R) -> GeneratorResult {
+        let connection_var = builder.get_or_create_random_connection(rng);
+
+        // Generate initial `LoadFilterLoad` operation
+        let mut filter_var = init_filter(builder, rng);
+
+        if rng.gen_bool(0.2) {
+            // Populate the filter with transactions or output data present in the program
+            let available_txs = builder.get_random_variables(rng, Variable::ConstTx);
```

Comment:
One question about this
We just want to add two random tx or txos here? (not everything?)
maybe because adding everything is too costly?